java/adapter/avro/src/test/java/org/apache/arrow/AvroTestBase.java

```diff
@@ -0,0 +1,178 @@
+    for (int i = 0; i < data.size(); i++) {
+      Object value1 = data.get(i);
+      Object value2 = vector.getObject(i);
+      if (value1 == null) {
+        assertTrue(value2 == null);
+        continue;
+      }
+      if (value2 instanceof byte[]) {
+        value2 = ByteBuffer.wrap((byte[]) value2);
+        if (value1 instanceof byte[]) {
+          value1 = ByteBuffer.wrap((byte[]) value1);
+        }
+      } else if (value2 instanceof Text) {
+        value2 = value2.toString();
+      }
+      assertTrue(Objects.equals(value1, value2));
+    }
+  }
+
+  protected void checkRecordResult(Schema schema, ArrayList<GenericRecord> data, VectorSchemaRoot root) {
+    assertEquals(data.size(), root.getRowCount());
+    assertEquals(schema.getFields().size(), root.getFieldVectors().size());
+
+    for (int i = 0; i < schema.getFields().size(); i++) {
+      ArrayList fieldData = new ArrayList();
+      for (GenericRecord record : data) {
+        fieldData.add(record.get(i));
+      }
+
+      checkPrimitiveResult(fieldData, root.getFieldVectors().get(i));
+    }
+
+  }
+
+
+  // belows are for iterator api
+
+  protected void checkArrayResult(List<List> expected, List<ListVector> vectors) {
+    int index = 0;
+    for (ListVector vector : vectors) {
+      for (int i = 0; i < vector.getValueCount(); i++) {
+        checkArrayElement(expected.get(index++), (JsonStringArrayList) vector.getObject(i));
+      }
+    }
+  }
+
+  protected void checkRecordResult(Schema schema, ArrayList<GenericRecord> data, List<VectorSchemaRoot> roots) {
+    roots.forEach(root -> {
+      assertEquals(schema.getFields().size(), root.getFieldVectors().size());
+    });
+
+    for (int i = 0; i < schema.getFields().size(); i++) {
+      List fieldData = new ArrayList();
+      List<FieldVector> vectors = new ArrayList<>();
+      for (GenericRecord record : data) {
+        fieldData.add(record.get(i));
+      }
+      final int columnIndex = i;
+      roots.forEach(root -> vectors.add(root.getFieldVectors().get(columnIndex)));
+
+      checkPrimitiveResult(fieldData, vectors);
+    }
+
+  }
+
+  protected void checkPrimitiveResult(List data, List<FieldVector> vectors) {
+    int valueCount = vectors.stream().mapToInt(v -> v.getValueCount()).sum();
+    assertEquals(data.size(), valueCount);
+
+    int index = 0;
+    for (FieldVector vector : vectors) {
+      for (int i = 0; i < vector.getValueCount(); i++) {
+        Object value1 = data.get(index++);
+        Object value2 = vector.getObject(i);
+        if (value1 == null) {
+          assertTrue(value2 == null);
+          continue;
+        }
+        if (value2 instanceof byte[]) {
+          value2 = ByteBuffer.wrap((byte[]) value2);
+          if (value1 instanceof byte[]) {
+            value1 = ByteBuffer.wrap((byte[]) value1);
+          }
+        } else if (value2 instanceof Text) {
+          value2 = value2.toString();
+        }
+        assertTrue(Objects.equals(value1, value2));
```

Comment:
can't you use assertEquals?  The same applies to other places where you use this.